rust/otap-dataflow/crates/otap/src/debug_processor/output.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Handles the output flow of the debug processor
+
+use crate::pdata::{OtapPdata, OtlpProtoBytes};
+use otap_df_config::PortName;
+use otap_df_engine::error::Error;
+use otap_df_engine::local::processor as local;
+use otap_df_engine::node::NodeId;
+use otel_arrow_rust::proto::opentelemetry::{
+    common::v1::{AnyValue, InstrumentationScope},
+    logs::v1::{LogRecord, LogsData, ResourceLogs, ScopeLogs, SeverityNumber},
+    resource::v1::Resource,
+};
+use prost::Message as _;
+use serde::Deserialize;
+use tokio::fs::File;
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(untagged)]
+pub enum OutputMode {
+    Console,
+    Outport(Vec<PortName>),
+    File(String),
+}
+
+/// struct that handles the logic for sending data to console or out_ports
+pub struct DebugOutput {
+    writer: Option<OutputWriter>,
+    ports: Option<Vec<PortName>>,
+    effect_handler: local::EffectHandler<OtapPdata>,
+}
+
+/// A wrapper around AsyncWrite that simplifies error handling for debug output
+pub struct OutputWriter {
+    writer: Box<dyn AsyncWrite + Unpin>,
+    processor_id: NodeId,
+}
+
+impl OutputWriter {
+    pub async fn new(output_file: Option<String>, processor_id: NodeId) -> Result<Self, Error> {
+        let writer: Box<dyn AsyncWrite + Unpin> = match output_file {
+            Some(file_name) => {
+                let file = File::options()
+                    .write(true)
+                    .append(true)
+                    .create(true)
+                    .open(file_name)
+                    .await
+                    .expect("could not open output file");
```

Comment:
A good principle will be be to emit a structured error, so the instantiation of the DebugProcessor will fail and the error will be properly reported.